homeassistant/data_entry_flow.py

```diff
@@ -311,7 +321,47 @@ async def async_configure(
         if (
             data_schema := cur_step.get("data_schema")
         ) is not None and user_input is not None:
-            user_input = data_schema(user_input)
+            try:
+                user_input = data_schema(user_input)
+            except vol.Invalid as ex:
+                raised_errors = [ex]
+                if isinstance(ex, vol.MultipleInvalid):
+                    raised_errors = ex.errors
+
+                schema_errors: dict[str, Any] = {}
+                for error in raised_errors:
+                    if error.path:
+                        current_path_schema = data_schema.schema
+                        current_path_errors = schema_errors
+                        for path_part in error.path[:-1]:
```

Comment:
Please invert the logic:

```python
if not error.path:
  ...
...
```

The code below is getting quite complex, please break it up in smaller parts,

What is happening with error's with a falsy path?

Reply:
I have extracted the error handling in a new function, and should be cleaner now. What do you think?